From 1d417d4b0d2ca6f084b14e527224fa44d32c4d75 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:26 -0400
Subject: [cpufreq] p4-clockmod: use cpufreq_table_validate_and_show()

Message-id: <1409235879-1283-53-git-send-email-prarit@redhat.com>
Patchwork-id: 89448
O-Subject: [RHEL7.1 PATCH BZ 1134639 052/185] cpufreq: p4-clockmod: use cpufreq_table_validate_and_show()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 719ffe495cfb86492ab47b05fde55fd5eafec159
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Mon Sep 16 18:56:24 2013 +0530

    cpufreq: p4-clockmod: use cpufreq_table_validate_and_show()

    Lets use cpufreq_table_validate_and_show() instead of calling
    cpufreq_frequency_table_cpuinfo() and cpufreq_frequency_table_get_attr().

    Cc: David S. Miller <davem@davemloft.net>
    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/p4-clockmod.c b/drivers/cpufreq/p4-clockmod.c
index 2f0a2a6..03478bf 100644
--- a/drivers/cpufreq/p4-clockmod.c
+++ b/drivers/cpufreq/p4-clockmod.c
@@ -230,7 +230,6 @@ static int cpufreq_p4_cpu_init(struct cpufreq_policy *policy)
   else
    p4clockmod_table[i].frequency = (stock_freq * i)/8;
  }
- cpufreq_frequency_table_get_attr(p4clockmod_table, policy->cpu);
 
  /* cpuinfo and default policy values */
 
@@ -239,7 +238,7 @@ static int cpufreq_p4_cpu_init(struct cpufreq_policy *policy)
  policy->cpuinfo.transition_latency = 10000001;
  policy->cur = stock_freq;
 
- return cpufreq_frequency_table_cpuinfo(policy, &p4clockmod_table[0]);
+ return cpufreq_table_validate_and_show(policy, &p4clockmod_table[0]);
 }
 
 
-- 
1.7.1